### PR TITLE
feat: Duplicate experiment key issue with multi feature flag

### PR DESCRIPTION
**Pull request by ozayr-zaviar**
## Summary
- Fixed an issue related to duplicate experimentKey.
- While trying to get variation from the variationKeyMap, it was unable to find because the latest experimentKey was overriding the previous one.

## TestPlan
- Run with FSC

### CODE REVIEW

File: src/Optimizely/Optimizely.php [hunk 1]
```diff
@@ -360,9 +362,11 @@ public function decide(OptimizelyUserContext $userContext, $key, array $decideOp
             $variationKey = $variation->getKey();
             $featureEnabled = $variation->getFeatureEnabled();
             $ruleKey = $decision->getExperiment()->getKey();
+            $experimentId = $decision->getExperiment()->getId();
         } else {
             $variationKey = null;
             $ruleKey = null;
+            $experimentId = null;
```
Comment (zashraf1985) [inline src/Optimizely/Optimizely.php:369]: this looks redundant?

**Review by zashraf1985 — APPROVED** [review]
Looks Good! Just added a comment to fix a redundant variable assignment.

**Comment by yasirfolio3** [comment]
LGTM!

**Review by jaeopt — CHANGES_REQUESTED** [review]
All changes look good. A couple of suggestions -
1. can we add new test cases for duplicate keys (duplicate experiment keys, and/or duplicate delivery-rule keys)
2. I found this line is still vulnerable to duplicate keys. Can you confirm it's ok? https://github.com/optimizely/php-sdk/blob/14e336c895037d5b3629b002faf2ea2b0e021054/src/Optimizely/DecisionService/DecisionService.php#L599

**Comment by msohailhussain** [comment]
> All changes look good. A couple of suggestions -
> 
> 1. can we add new test cases for duplicate keys (duplicate experiment keys, and/or duplicate delivery-rule keys)
> 2. I found this line is still vulnerable to duplicate keys. Can you confirm it's ok? https://github.com/optimizely/php-sdk/blob/14e336c895037d5b3629b002faf2ea2b0e021054/src/Optimizely/DecisionService/DecisionService.php#L599

Thanks Jae. Will take care of it.

**Review by jaeopt — APPROVED** [review]
LGTM
